buildSrc/src/main/kotlin/publish-conventions.gradle.kts

```
@@ -0,0 +1,75 @@
+plugins {
+    `maven-publish`
+    signing
+}
+
+// empty xxx-javadoc.jar
+val javadocJar by tasks.register<Jar>("javadocJar") {
+    archiveClassifier.set("javadoc")
+}
+
+publishing {
+    repositories {
+        maven {
+            url = mavenRepositoryUri()
+            credentials {
+                username = project.getSensitiveProperty("libs.sonatype.user")
+                password = project.getSensitiveProperty("libs.sonatype.password")
+            }
+        }
+    }
+
+    publications.withType<MavenPublication>().configureEach {
+        pom {
+            name = project.name
+            description = "AtomicFU utilities"
+            url = "https://github.com/Kotlin/kotlinx.atomicfu"
+
+            licenses {
+                license {
+                    name = "The Apache Software License, Version 2.0"
+                    url = "https://www.apache.org/licenses/LICENSE-2.0.txt"
+                    distribution = "repo"
+                }
+            }
+
+            developers {
+                developer {
+                    id = "JetBrains"
+                    name = "JetBrains Team"
+                    organization = "JetBrains"
+                    organizationUrl = "https://www.jetbrains.com"
+                }
+            }
+
+            scm {
+                url = "https://github.com/Kotlin/kotlinx.atomicfu"
+            }
+        }
+
+
+        signPublicationIfKeyPresent(project, this)
+        // add empty javadocs
+        if (name != "kotlinMultiplatform") { // The root module gets the JVM's javadoc JAR
+            artifact(javadocJar)
+        }
+    }
+
+    tasks.withType<AbstractPublishToMaven>().configureEach {
+        mustRunAfter(tasks.withType<Sign>())
+    }
+
+    // NOTE: This is a temporary WA, see KT-61313.
+    tasks.withType<Sign>().configureEach sign@{
+        val pubName = name.substringAfter("sign").substringBefore("Publication")
+
+        // Task ':linkDebugTest<platform>' uses this output of task ':sign<platform>Publication' without declaring an explicit or implicit dependency
+        tasks.findByName("linkDebugTest$pubName")?.let { linkTask ->
+            linkTask.mustRunAfter(this@sign)
+        }
+        // Task ':compileTestKotlin<platform>' uses this output of task ':sign<platform>Publication' without declaring an explicit or implicit dependency
+        tasks.findByName("compileTestKotlin$pubName")?.let { compileTask ->
+            compileTask.mustRunAfter(this@sign)
+        }
+    }
```

Continued from https://github.com/Kotlin/kotlinx-atomicfu/pull/435#discussion_r1598070977

```suggestion
    // NOTE: This is a temporary WA, see KT-61313.
    // Task ':linkDebugTest<platform>' uses this output of task ':sign<platform>Publication' without declaring an explicit or implicit dependency
    tasks.withType<AbstractPublishToMaven>().configureEach {
        mustRunAfter(tasks.withType<Sign>())
    }
```

Let's use a simpler workaround. Even if it's less efficient it's easier to understand & maintain.